src/realm/transaction.cpp

```diff
@@ -52,6 +52,9 @@ void generate_properties_for_obj(Replication& repl, const Obj& obj, const ColInf
         auto embedded_table = elem.second;
         auto cols_2 = get_col_info(embedded_table);
         auto update_embedded = [&](Mixed val) {
+            if (val.is_null()) {
```

Comment:
so it seems we never really cater for this use case, we have always taken for granted the fact that the link could have been NULL, I think this is the result of how we are masking invalid links. So I think this fix is correct.